### ISSUE TITLE
[BUG]: Empty or no provider information leads to error

### ISSUE THREAD

**Issue by schlichtanders** [integrations/terraform-provider-github#2008]
### Expected Behavior

the github integration should work anonymously without providing any provider information

### Actual Behavior

with no `provider "github" {}` I get the following error on the most recent version
```
│ Error: Invalid provider configuration
│ 
│ Provider "registry.terraform.io/integrations/github" requires explicit configuration. Add a provider block to the root module and configure the provider's required
│ arguments as described in the provider documentation.
│ 
╵
╷
│ Error: GET https://api.github.com/user: 401 Requires authentication []
│ 
│   with provider["registry.terraform.io/integrations/github"],
│   on <empty> line 0:
│   (source code not available)
│ 
```

with setting the empty `provider "github" {}`, I still get 
```
╷
│ Error: GET https://api.github.com/user: 401 Requires authentication []
│ 
│   with provider["registry.terraform.io/integrations/github"],
│   on kube.tf line 754, in provider "github":
│  754: provider "github" {}
│ 
```

### Terraform Version

Terraform v1.5.3
on linux_amd64

Your version of Terraform is out of date! The latest version
is 1.6.3. You can update by downloading from https://www.terraform.io/downloads.html

### Affected Resource(s)

- [kube-hetzner](https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner)
- probably others as well

### Steps to Reproduce

you could create a kubernetes cluster using https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner and see that it fails, but I guess that is not yet minimal enough.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by rcambrj** [comment]
This appears to have been introduced in [v5.41.0..v5.42.0](https://github.com/integrations/terraform-provider-github/compare/v5.41.0...v5.42.0)

**Comment by mysticaltech** [comment]
Waiting for the fix too.

**Comment by mnencia** [comment]
Solved in release 5.44.0.

@kfcampbell please close this issue.

**Comment by mabilinab** [comment]
Hey! I'm getting this error again. Any tips?
